debian/docker.io.install

```diff
@@ -1,12 +1,12 @@
 #!/usr/bin/dh-exec
 
 # https://github.com/docker/docker-ce/blob/v19.03.2/components/packaging/deb/common/rules#L36
-components/cli/build/docker => usr/bin/docker
-components/engine/bundles/dynbinary-daemon/dockerd => usr/bin/dockerd
+cli/build/docker => usr/bin/docker
+engine/bundles/dynbinary-daemon/dockerd => usr/bin/dockerd
 libnetwork/docker-proxy => usr/bin/docker-proxy
 tini/build/tini-static => usr/bin/docker-init
 
-components/cli/contrib/completion/zsh/_docker usr/share/zsh/vendor-completions/
-components/engine/contrib/check-config.sh usr/share/docker.io/contrib/
-components/engine/contrib/nuke-graph-directory.sh usr/share/docker.io/contrib/
-components/packaging/systemd/docker.socket lib/systemd/system/
+cli/contrib/completion/zsh/_docker usr/share/zsh/vendor-completions/
+engine/contrib/check-config.sh usr/share/docker.io/contrib/
+engine/contrib/nuke-graph-directory.sh usr/share/docker.io/contrib/
+debian/docker.socket lib/systemd/system/
```

Comment:
I *think* if we name it `debian/docker.io.docker.socket` that debhelper will now do the right thing by itself (without this line for `dh_install`) -- it appears this is documented in `dh_installsystemduser`. :innocent:

Reply:
I tried to rename `debian/docker.socket` to what you proposed and removed that line from `debian/docker.io.install` but at the end the socket file was not installed in the binary package.

And the `dh_installsystemduser` installs files in `usr/lib/systemd/user/ ` not in `lib/systemd/system/` as the current package does. I believe this is not what we want to do.

Reply:
What we want is `dh_installsystemd` but in its manpage it says that it works with `debian/package.socket` and `debian/package@.socket`, I do not know whether this syntax works for socket files (`<package_name>.<socket_file>`).

Using `debian/docker.io.socket` installs the file in the right place but it is named `docker.io.socket` instead of `docker.socket` (what we have in the previous versions).

Reply:
I am not sure why but this solution is not working for me :/

Reply:
Oh, I think we need the following, if we don't have it already:

```make
override_dh_installsystemd:
	dh_installsystemd --name=docker
```

Reply:
(Which is what tells it to look for `docker.io.docker.socket`, etc)

Reply:
The change you proposed worked, but it required a bump debhelper-compat level to 11, otherwise this feature would no be available. I believe this is fine since compat level 11 is available from Bionic on which allows us to backport it if we decide to do so in the future.